server/src/main/proto/spine/server/aggregate/aggregate.proto

```diff
@@ -34,73 +33,80 @@ import "google/protobuf/timestamp.proto";
 
 import "spine/annotations.proto";
 import "spine/base/event.proto";
+import "spine/base/version.proto";
 
-// Built-in event type for storing aggregate snapshots.
+// The message for storing an aggregate snapshot.
 message Snapshot {
-    option (SPI_type) = true;
 
     // The state of the aggregate.
     google.protobuf.Any state = 1;
 
-    // Time when the stored state of the aggregate was modified.
-    //
-    // This value is copied to the aggregated when the snapshot is restored.
-    google.protobuf.Timestamp when_modified = 2;
-
-    // The version of the aggregate at which the snapshot was made.
-    int32 version = 3;
+    // The version of the aggregate.
+    spine.base.Version version = 2;
 
     // The moment the snapshot was created.
     //
-    // This value should normally be after the time the aggregate was modified (and timestamp of the
-    // corresponding event), and before the time of the next event in the aggregate history.
-    google.protobuf.Timestamp timestamp = 4;
+    // This value should normally be after the time the aggregate was modified
+    // (and timestamp of the corresponding event), and before the time of
+    // the next event in the aggregate history.
+    //
+    // This field is used when storing a snapshot in `AggregateEventRecord`.
+    //
+    google.protobuf.Timestamp timestamp = 3;
 }
 
-// A record in the storage of events and snapshots of an aggregate type.
-//
-// Different types of aggregates are managed by different storages.
-//
-message AggregateStorageRecord {
-    option (SPI_type) = true;
+// A record with information about an event or a snapshot of an aggregate.
+message AggregateEventRecord {
 
     // A timestamp of the event or snapshot in this record.
     google.protobuf.Timestamp timestamp = 1;
 
     // A short Protobuf type name of the event.
+    //
+    // Since events stored are for particular aggregate type, we are not likely to
+    // have a name clash because of not storing fully-qualified names.
+    //
+    // This field is blank if the record stores a snapshot.
+    //
     string event_type = 2;
 
     // String representation of `EventId`.
+    //
+    // When storing a snapshot this field contains the string `Snapshot`.
```

Comment:
Let's discuss together with @dmdashenkov. This and the above field look like kind of optimization for storage, which we dropped already for other record types.